core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -282,6 +282,7 @@ class TaskData private[spark](
     val speculative: Boolean,
     val accumulatorUpdates: Seq[AccumulableInfo],
     val errorMessage: Option[String] = None,
+    val failureReason: Option[FailureReason] = None,
```

Comment:
Existing `errorMessage` and newly added `failureReason` are almost the same.
![redundant-error-messages](https://user-images.githubusercontent.com/4736016/93766864-1058e300-fc52-11ea-8b44-fe99b5283003.png)
Can we avoid adding `failureReason` by building `FailureReason` in `computeFailureSummary`?

Reply:
Answered in above comments.